### PR TITLE
Microsoft.Insights Log Analytics Query Pack Resource

### PR DESCRIPTION
**Pull request by arolshan**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+          "description": "Resource Types associated with the query.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "LogAnalyticsQueryPackQueryListResult": {
+      "description": "Describes the list of Log Analaytics QueryPack-Query resources.",
```
Comment (nschonni) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]:
```suggestion
      "description": "Describes the list of Log Analytics QueryPack-Query resources.",
```

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A list containing 0 or more Log Analaytics QueryPack definitions.",
```
Comment (nschonni) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]:
```suggestion
            "description": "A list containing 0 or more Log Analytics QueryPack definitions.",
```

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Current state of this QueryPack: whether or not is has been provisioned within the resource group it is defined. Users cannot change this value but are able to read from it. Values will include Succeeded, Deploying, Canceled, and Failed.",
+          "readOnly": true
+        }
+      }
+    },
+    "LogAnalyticsQueryPackListResult": {
+      "description": "Describes the list of Log Analaytics QueryPack resources.",
```
Comment (nschonni) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]:
```suggestion
      "description": "Describes the list of Log Analytics QueryPack resources.",
```

**Review by nschonni — COMMENTED** [review]
Analaytics -> Analytics

**Review by anuchandy — CHANGES_REQUESTED** [review]
Please take a look at model validation errors https://dev.azure.com/azure-sdk/public/_build/results?buildId=91363

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 3 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: fixed, thanks

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 2 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: fixed thanks

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 1 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: fixed thanks

**Comment by arolshan** [comment]
> Please take a look at model validation errors https://dev.azure.com/azure-sdk/public/_build/results?buildId=91363

fixed

**Comment by anuchandy** [comment]
@arolshan thanks for fixing the errors, I've pinged ARM team for the review.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+        "operationId": "QueryPacks_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: For each API/method, please include a "default" response that handles all error cases. A common definition you can reference is here: https://github.com/Azure/azure-rest-api-specs/blob/c136fa74ed9b0fb69be01cce6c94fb44255029f0/specification/common-types/resource-management/v1/types.json#L272

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+          "description": "Creation Date for the Log Analytics QueryPack, in ISO 8601 format.",
+          "format": "date-time"
+        },
+        "timeModified": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Last modified date of the Log Analytics QueryPack, in ISO 8601 format.",
+          "format": "date-time"
+        },
+        "tenantId": {
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: Why persist this? Do you update it if the subscription is moved? This is always resolvable from the subscriptionId

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure tenant id."
+        },
+        "subscriptionId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Subscription id."
+        },
+        "resourceGroup": {
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: subscriptionId and resourceGroup are in the resourceId, which is returne on each response. They are redundant in the resource properties

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+      "required": ["properties"],
+      "description": "An Log Analytics QueryPack definition."
+    },
+    "LogAnalyticsQueryPackProperties": {
+      "description": "Properties that define a Log Analytics QueryPack resource.",
+      "properties": {
+        "queryPackId": {
+          "type": "string",
+          "description": "The unique ID of your application. This field cannot be changed.",
+          "readOnly": true
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: This is generated by the service?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+        },
+        "name": {
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: Is this different from the top-level name property? If so, a name like displayName might be more appropriate. If it is the same as the top-level name property, it's redundant and should be removed. Does it really need to be unique?

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 7 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: yes

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 5 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: thanks, fixed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 6 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: thanks, your'e right, fixed.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 8 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: thanks, your'e right, removed. resource name is unique within resource group. uniqueness is achieved by resource id.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 5 above)
Comment (pilor) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: Kris's comment was about the "tenantId" here.  Is there a reason it needs to be stored with the resource and not resolved from the subscription registration?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 9]
```diff
@@ -0,0 +1,464 @@
...
+        }
+      }
+    },
+    "QueriesResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: what about name and type?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 10]
```diff
@@ -0,0 +1,464 @@
...
+        "properties": {
+          "$ref": "#/definitions/LogAnalyticsQueryPackQueryProperties"
+        }
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: there is a bit of a circular reference going on here.  The "QueriesResource" contains QueryPackQueryProperties but the LogAnalyticsQueryPackQuery contains QueriesResource AND its own QueryPackQueryProperties

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 11]
```diff
@@ -0,0 +1,464 @@
...
+      "properties": {
+        "queryId": {
+          "type": "string",
+          "description": "The unique ID of your application. This field cannot be changed.",
+          "readOnly": true
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Unique name for your Query Pack."
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >Unique name [](start = 26, length = 11)

is this truly unique?  If so, at what scope?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 12]
```diff
@@ -0,0 +1,464 @@
...
+        },
+        "timeCreated": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Creation Date for the Log Analytics QueryPack, in ISO 8601 format.",
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >Log Analytics QueryPack [](start = 48, length = 23)

Is timeCreated/timeModified here the dat of the QueryPack creation/modification as written or the date of the query creation/modification?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 13]
```diff
@@ -0,0 +1,464 @@
...
+        },
+        "author": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Object Id of user creating the query."
+        },
+        "description": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Azure tenant id."
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >"Azure tenant id. [](start = 25, length = 17)

Update

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 14]
```diff
@@ -0,0 +1,464 @@
...
+        },
+        "body": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Subscription id."
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >Subscription id. [](start = 26, length = 16)

Update

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 15]
```diff
@@ -0,0 +1,464 @@
...
+        },
+        "linkedResourceId": {
+          "type": "string",
+          "readOnly": false,
+          "description": "Resource id associated with the query."
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >Resource id associated with the query. [](start = 26, length = 38)

What is this linked resource used for?  How is it related to the query?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 16]
```diff
@@ -0,0 +1,464 @@
...
+        },
+        "resourceTypes": {
+          "description": "Labels associated with the query.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "labels": {
+          "description": "Resource Types associated with the query.",
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: descriptions are swapped

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 17]
```diff
@@ -0,0 +1,464 @@
...
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: descriptions are swapped

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 18]
```diff
@@ -0,0 +1,464 @@
...
+    "IncludeContentParameter": {
+      "name": "includeContent",
+      "in": "query",
+      "required": false,
+      "type": "boolean",
+      "description": "Flag indicating whether or not to return the content of each applicable query. If false, only return the query information.",
+      "x-ms-parameter-location": "method"
+    },
+    "MaxItemCountParameter": {
+      "name": "maxItemCount",
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >maxItemCount [](start = 15, length = 12)

this should be $top according to the RPC ([https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3))

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 19]
```diff
@@ -0,0 +1,464 @@
...
+      "in": "query",
+      "required": false,
+      "type": "integer",
+      "description": "Maximum items returned in page.",
+      "x-ms-parameter-location": "method"
+    },
+    "ContinuationTokenParameter": {
+      "name": "continuationToken",
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >continuationToken [](start = 15, length = 17)

this should be $skipToken according to the RPC ([https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#response-3))

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 20]
```diff
@@ -0,0 +1,464 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "A list containing 0 or more queries associated with the Log Analytics QueryPack.",
+            "schema": {
+              "$ref": "#/definitions/LogAnalyticsQueryPackQueryListResult"
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >LogAnalyticsQueryPackQueryListResult [](start = 37, length = 36)

any operation returning this definition should be marked as x-ms-pageable ([https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable))

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesSearch.json [hunk 21]
```diff
@@ -0,0 +1,65 @@
+{
+  "parameters": {
+    "api-version": "2019-09-01-preview",
+    "subscriptionId": "86dc51d3-92ed-4d7e-947a-775ea79b4918",
+    "resourceGroupName": "my-resource-group",
+    "queryPackName": "my-querypack",
+    "includeContent": true,
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesSearch.json]: >"includeContent": true, [](start = 4, length = 23)

what additional content is included or excluded by this parameter?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json (hunk 5 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPacks_API.json]: for now, your'e right, there's no need to explicitly contain it in the resource.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 9 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks, added.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 10 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: yes youre right, fixed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 11 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: the scope is the query pack.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 12 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: the query.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 13 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks !

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 15 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 16 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks !

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesSearch.json (hunk 21 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesSearch.json]: I changed this to **includeBody**. 
According to our experience the body of the query can get quite large, so to decrease the size of the payload for efficiency we allow this parameter. In addition, in some flow for UI purposes all that is needed is metadata of the query and not the body itself.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 20 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: done

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 18 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: done

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 19 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: done

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 12 above)
Comment (pilor) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: The description needs to be updated then.  It is stating it is the creation time of the parent query pack

---
In reply to: [321102457](https://github.com/Azure/azure-rest-api-specs/pull/6992#discussion_r321102457) [](ancestors = 321102457)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 17 above)
Comment (pilor) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: still incorrect

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json [hunk 22]
```diff
@@ -0,0 +1,477 @@
...
+    },
+    "QueryIdParameter": {
+      "name": "queryId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The id of a specific query defined in the Log Analytics QueryPack",
+      "x-ms-parameter-location": "method"
+    },
+    "IncludeBodyParameter": {
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: >"IncludeBodyParameter" [](start = 4, length = 22)

The default if this parameter is not included is to include the body, correct?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json [hunk 23]
```diff
@@ -0,0 +1,42 @@
+{
+  "parameters": {
+    "api-version": "2019-09-01-preview",
+    "subscriptionId": "86dc51d3-92ed-4d7e-947a-775ea79b4918",
+    "resourceGroupName": "my-resource-group",
+    "queryPackName": "my-querypack",
+    "queryId": "a449f8af-8e64-4b3a-9b16-5a7165ff98c4",
+    "QueryPayload": {
+      "name": "a449f8af-8e64-4b3a-9b16-5a7165ff98c4",
+      "type": "microsoft.insights/querypacks/queries",
```
Comment (pilor) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json]: I think your fix for the circular reference messed things up.  "name" and "type" should not be included here.  Name comes from queryPackName and type is readonly

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 22 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: exactly

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 17 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: sorry, fixed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json (hunk 23 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json]: fixed.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 12 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks !

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json (hunk 23 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/examples/QueryPackQueriesPut.json]: fixed

File: specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json (hunk 12 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-09-01-preview/QueryPackQueries_API.json]: thanks !

**Comment by pilor** [comment]
LGTM

**Review by pilor — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**